### DISCUSSION TITLE
`Server has closed the connection` when usnig two PrismaClients

### DISCUSSION THREAD

**Discussion by rostamiani** [prisma/prisma#16490] [Q&A]
I have a project with Prisma+NestJS. My database is a master+replication PostgresSQL server with pgBouncer
It's executed in local server but `Server has closed the connection` happened for some customers when published. This error happened for different tables. This is just an example:
```
Invalid `this.prisma.readClient.customer.findUnique()` invocation in /usr/src/app/dist/src/modules/global/guards/auth-guard.js
54 const customer = await this.prisma.readClient.customer.findUnique( 
Server has closed the connection.
```

Connection Strings look like this:
```
DATABASE_WRITE_URL="postgresql://user:******@[masterip]:5432/database?schema=public&pgbouncer=true"
DATABASE_READ_URL="postgresql://user:******:@[replicaip]:5432/stage-v2-panel?schema=public&pgbouncer=true"
```

PrismaService file:
```ts
@Injectable()
export class PrismaService implements OnModuleInit {
  readClient: PrismaClient;
  writeClient: PrismaClient;

  constructor() {
    this.readClient = new PrismaClient({
      datasources: {db: {url: process.env.DATABASE_READ_URL}},
    });
    this.writeClient = new PrismaClient({
      datasources: {db: {url: process.env.DATABASE_WRITE_URL}},
    });
  }

  async onModuleInit() {
    await this.readClient.$connect();
    await this.writeClient.$connect();
  }

  async enableShutdownHooks(app: INestApplication) {
    this.readClient.$on('beforeExit', async () => {
      await app.close();
    });
    this.writeClient.$on('beforeExit', async () => {
      await app.close();
    });
  }
}
```

**Comment by nurul3101** [comment]
Hello @rostamiani 👋 

So if I understand, this error happens intermittently, right?
Have you configured any [connection_limit](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#increasing-the-pool-size) parameters in your database connection strings?

**Reply by vladi-stevanovic** [reply]
👋 Hi @rostamiani , did you perhaps have a chance to check Nurul's questions? Let us know if you still need help!

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by rostamiani** [reply]
@nurul3101 
Connection limit is 2000! but clients use a PgBouncer server with a pool with 600 connections
When using a single connection string, there is no problem
@vladi-stevanovic Not yet... We are working on.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
